### PR TITLE
rename stored job namedtuple/enum => instigation

### PR DESCRIPTION
**Pull request by prha**
## Summary
The serialized schedule/sensor information uses the name `Job`, which is confusing now that it has significance in the repository definition sense.

This renames all of the data classes / named tuples / enums to use the word "instigation".

## Test Plan
BK

### CODE REVIEW

File: python_modules/dagster/dagster/core/definitions/run_request.py [hunk 1]
```diff
@@ -3,16 +3,21 @@
...
-from dagster.serdes import whitelist_for_serdes
+from dagster.serdes.serdes import register_serdes_enum_fallbacks, whitelist_for_serdes
 from dagster.utils.error import SerializableErrorInfo
 
 
 @whitelist_for_serdes
-class JobType(Enum):
+class InstigationType(Enum):
     SCHEDULE = "SCHEDULE"
     SENSOR = "SENSOR"
```
Comment (gibsondan) [inline python_modules/dagster/dagster/core/definitions/run_request.py:13]: Isn't the Schedule the Instigator not the Instigation?

File: python_modules/dagster/dagster_tests/scheduler_tests/test_scheduler_timezones.py [hunk 2]
```diff
@@ -746,7 +746,7 @@ def test_execute_during_dst_transition_spring_forward(external_repo_context):
                     ticks[i],
                     external_schedule,
                     expected_datetimes_utc[i],
-                    JobTickStatus.SUCCESS,
+                    InstigationTickStatus.SUCCESS,
```
Comment (gibsondan) [inline python_modules/dagster/dagster_tests/scheduler_tests/test_scheduler_timezones.py:749]: Consider just TickStatus, not sure the Instigation is needed here

**Review by gibsondan — APPROVED** [review]
this is very bikesheddy but did you consider Instigator not Instigation? InstigatorStatus / InstigatorState / etc.

The Schedule is the thing that instigates runs right?  Instigation makes me think its referring to the runs / the things that were instigated, not the thing doing the instigating. Not a blocking comment though

**Comment by prha** [comment]
Yeah, I can redo all these names.  I agree that Instigator sounds better.

**Pull request merged**